spec.bs

```diff
@@ -130,110 +131,91 @@ are:
 
-Immediately after an operation completes, [=process the Shared Storage
-contributions cache=] given the worklet's [=contributions cache=] and the
-worklet's global scope.
+    Issue: Consider improving developer ergonomics here (e.g. a way to detect
+        this case).
 
-Issue: How to handle batching different operation invocations properly. This
-    doesn't work for simultaneous operations. Also disallow usage outside an
-    operation.
+1. Let |entry| be a new [=contribution cache entry=] with the items:
+    : [=contribution cache entry/contribution=]
+    :: |contribution|
+    : [=contribution cache entry/batching scope=]
+    :: The result of running |batchingScopeSteps|
 
-To <dfn>process the Shared Storage contributions cache</dfn> given a [=list=]
-|contributionsCache| and a {{SharedStorageWorkletGlobalScope}} |scope|, perform
-the [=PrivateAggregation/report creation and scheduling steps=] with |scope|'s
-[=relevant settings object=]'s [=environment settings object/origin=],
-"<code>[=context type/shared-storage=]</code>" and |contributionsCache|.
+1. [=Append an entry to the contribution cache=] given |entry|.
 
+Issue: Check that value can actually be zero in the spec pipeline.
 
-Exposing to the Protected Audience API {#protected-audience}
-============================================================
-
-<xmp class="idl">
-partial interface InterestGroupScriptRunnerGlobalScope {
-  readonly attribute PrivateAggregation privateAggregation;
-};
-
-dictionary PASignalValue {
-  required DOMString baseValue;
-  double scale;
-  (bigint or long) offset;
-};
-
-dictionary PAExtendedHistogramContribution {
-  required (PASignalValue or bigint) bucket;
-  required (PASignalValue or long) value;
-};
-
-[Exposed=InterestGroupScriptRunnerGlobalScope, SecureContext]
-partial interface PrivateAggregation {
-  undefined contributeToHistogramOnEvent(
-      DOMString event, PAExtendedHistogramContribution contribution);
-};
-</xmp>
-
-Issue: Do we want to align naming with implementation?
-
-Immediately after an auction completes, [=process the Protected Audience
-contributions cache=] given the worklet's [=contributions cache=] and the
-worklet's global scope.
-
-Issue: Does Protected Audience API have one global scope per auction or
-    multiple? If multiple, will need to change scope for batching.
-
-Issue: How to handle fenced frame-triggered contributions and other
-    event-triggered contributions.
-
-Issue: Need to handle `auctionReportBuyers` and `auctionReportBuyerKeys` here or
-    in the Protected Audience API spec.
-
-<div algorithm>
-The <dfn method for="PrivateAggregation">contributeToHistogramOnEvent(DOMString
-event, PAExtendedHistogramContribution contribution)</dfn> method steps are:
+Issue(44): Consider accepting an array of contributions.
+
+Exposing to global scopes {#exposing}
+=====================================
+
+To expose this API to a global scope, a readonly attribute of type
```

Comment:
Probably can make it a bit more precise like `a readonly attribute privateAggregation of type {{Private Aggregation}}`

Reply:
Done